src/main/java/mods/railcraft/common/fluids/tanks/StandardTank.java

```diff
@@ -87,9 +94,10 @@ public boolean matchesFilter(FluidStack fluidStack) {
 
     @Override
     public int fillInternal(FluidStack resource, boolean doFill) {
-        if (!matchesFilter(resource))
```

Comment:
No I didn't. I put it where it needed to be to prevent any possibility of filling with the wrong fluid.